datafusion/physical-expr/src/hash_utils.rs

```diff
@@ -96,6 +96,12 @@ fn hash_array_primitive<T>(
     T: ArrowPrimitiveType,
     <T as arrow_array::ArrowPrimitiveType>::Native: HashValue,
 {
+    assert_eq!(
```

Comment:
this is the same assert copied from a few  lines down and makes me feel better about

```rust
                let value = unsafe { array.value_unchecked(i) };
```